Mathlib/NumberTheory/ModularForms/EisensteinSeries/Basic.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck, David Loeffler
+-/
+import Mathlib.NumberTheory.ModularForms.SlashInvariantForms
+import Mathlib.NumberTheory.ModularForms.CongruenceSubgroups
+
+/-!
+# Eisenstein Series
+
+## Main definitions
+
+* We define Eisenstein series of level `Γ(N)` for any `N : ℕ` and weight `k : ℤ` as the infinite sum
+  `∑' v : (Fin 2 → ℤ), (1 / (v 0 * z + v 1) ^ k)`, where `z : ℍ` and `v` ranges over all pairs of
+  coprime integers congruent to a fixed pair `(a, b)` modulo `N`.
+* We show that they define a slash invariant form of level `Γ(N)` and weight `k`.
+
+## References
+* [F. Diamond and J. Shurman, *A First Course in Modular Forms*][diamondshurman2005]
+-/
+
+noncomputable section
+
+open ModularForm UpperHalfPlane Complex Matrix
+
+open scoped MatrixGroups
+
+namespace EisensteinSeries
+
+variable (N : ℕ) (a : Fin 2 → ZMod N)
+
+section gammaSet_def
+
+/-- The set of pairs of coprime integers congruent to `a` mod `N`. -/
+def gammaSet := {v : Fin 2 → ℤ | (↑·) ∘ v = a ∧ IsCoprime (v 0) (v 1)}
```

Comment:
This means the same, right? I think it's much more common to omit the `·` in this context
```suggestion
def gammaSet := {v : Fin 2 → ℤ | (↑) ∘ v = a ∧ IsCoprime (v 0) (v 1)}
```